zcash_client_backend/src/fees.rs

```diff
@@ -25,29 +24,34 @@ pub mod zip317;
 /// A proposed change amount and output pool.
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct ChangeValue {
-    output_pool: ShieldedProtocol,
+    output_pool: PoolType,
     value: NonNegativeAmount,
     memo: Option<MemoBytes>,
 }
 
 impl ChangeValue {
     /// Constructs a new change value from its constituent parts.
-    pub fn new(
-        output_pool: ShieldedProtocol,
-        value: NonNegativeAmount,
-        memo: Option<MemoBytes>,
-    ) -> Self {
+    pub fn new(output_pool: PoolType, value: NonNegativeAmount, memo: Option<MemoBytes>) -> Self {
```

Comment:
This should return `Option<Self>` as @str4d noted in the original PR; transparent change with a non-None memo should be disallowed by construction.

Reply:
Alternately, rename this constructor to `shielded` and take a `ShieldedProtocol` argument instead of `PoolType`.